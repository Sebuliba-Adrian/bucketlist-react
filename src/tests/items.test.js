import React from 'react';
import renderer from 'react-test-renderer';
import Items from '../components/items/items';

const props = {
  selectedBucketlist: {
    id: 1,
    title: 'Selected Bucketlist',
    description: 'This is the selected bucketlist',
    created_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
    updated_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
  },
  items: [
    {
      id: 1,
      title: 'Test item',
      description: 'This is a test item',
      created_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
      updated_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
      status: false,
    },
    {
      id: 2,
      title: 'Test item 2',
      description: 'This is a test item 2',
      created_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
      updated_at: 'Wed, 20 Sep 2017 11:31:50 GMT',
      status: true,
    },
  ],
  request: jest.fn(),
};
it('renders correctly', () => {
  const items = renderer.create(<Items {...props} />).toJSON();
  expect(items).toMatchSnapshot();
});
