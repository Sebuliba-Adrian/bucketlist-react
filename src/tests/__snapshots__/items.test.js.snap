// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="container items-container"
  >
    <div
      className="card mt-3 mb-3"
    >
      <div
        className="card-block"
      >
        <div
          className="d-flex w-100 justify-content-between mb-2"
        >
          <h5 />
          <button
            className="btn btn-sm btn-cool-blue col-xs-12"
            data-target="#addItemModal"
            data-toggle="modal"
          >
            <span
              className="fa fa-plus pull-left"
            />
             Add Item
          </button>
        </div>
        <nav
          className="ml-4"
        >
          <ul
            className="pagination pagination-sm"
          >
            <li
              className="page-item disabled"
            >
              <a
                className="page-link"
                href="#"
                tabindex="-1"
              >
                Previous
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                href="#"
              >
                1
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                href="#"
              >
                2
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                href="#"
              >
                3
              </a>
            </li>
            <li
              className="page-item"
            >
              <a
                className="page-link"
                href="#"
              >
                Next
              </a>
            </li>
          </ul>
        </nav>
        <div>
          <h6
            className="card-text ml-4 pb-2 text-center text-cool-blue"
          >
            Items in 
            Selected Bucketlist
          </h6>
          <ul
            className="list-group list-group-flush"
          >
            <div>
              <a
                className="list-group-item list-group-item-action flex-column align-items-start"
                href=""
                onClick={[Function]}
              >
                <div
                  className="container"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-md-2 hidden-sm-down"
                    >
                      <div
                        className="text-center date-display"
                      >
                        <div
                          className="date-display-month text-white bg-cool-blue"
                        >
                           Sep
                        </div>
                        <div
                          className="date-display-day"
                        >
                          <h4>
                             20
                          </h4>
                        </div>
                        <div
                          className="date-display-year"
                        >
                           2017
                        </div>
                      </div>
                    </div>
                    <div
                      className="col-md-10 item-data"
                    >
                      <div
                        className="d-flex w-100 justify-content-between"
                      >
                        <p>
                          <small
                            className="text-muted"
                          >
                            Pending
                          </small>
                        </p>
                        <div>
                          <div
                            className="fa fa-lg fa-pencil-square-o mr-2"
                            data-backdrop="static"
                            data-keyboard="false"
                            data-target="#1"
                            data-toggle="modal"
                            onClick={[Function]}
                          />
                          <div
                            className="fa fa-lg fa-trash-o ml-2"
                            data-backdrop="static"
                            data-keyboard="false"
                            data-target="#deleteItemModel1"
                            data-toggle="modal"
                            onClick={[Function]}
                          />
                        </div>
                      </div>
                      <h5
                        className="mb-1 text-cool-blue"
                      >
                        Test item
                      </h5>
                      <h6
                        className="text-muted"
                      >
                        This is a test item
                      </h6>
                      <div>
                        <small
                          className="text-muted"
                        >
                          <strong>
                            Updated: 
                          </strong>
                           20 Sep 2017 11:31:50 GMT
                        </small>
                      </div>
                    </div>
                  </div>
                </div>
              </a>
              <div
                className="modal fade"
                id={1}
                role="dialog"
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <div
                      className="modal-header"
                    >
                      <h5
                        className="modal-title"
                        id="exampleModalLabel"
                      >
                        Edit a item
                      </h5>
                      <button
                        aria-label="Close"
                        className="close"
                        data-dismiss="modal"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                      </button>
                    </div>
                    <div
                      className="modal-body"
                    >
                      <label
                        htmlFor="tiltle"
                      >
                        Title
                      </label>
                      <div
                        className="input-group"
                        onChange={[Function]}
                      >
                        <input
                          className="form-control"
                          id="tiltle"
                          name="title"
                          onChange={[Function]}
                          type="text"
                          value="Test item"
                        />
                      </div>
                      <br />
                      <label
                        htmlFor="description"
                      >
                        Description
                      </label>
                      <div
                        className="input-group"
                      >
                        <textarea
                          className="form-control"
                          id="description"
                          name="description"
                          onChange={[Function]}
                          type="text"
                          value="This is a test item"
                        />
                      </div>
                      <br />
                      <div
                        className="form-group"
                      >
                        <label
                          htmlFor="status"
                        >
                          Status
                        </label>
                        <select
                          className="form-control"
                          id="status"
                          name="status"
                          onChange={[Function]}
                          value="0"
                        >
                          <option
                            value="1"
                          >
                            Finished
                          </option>
                          <option
                            value="0"
                          >
                            Pending
                          </option>
                        </select>
                      </div>
                    </div>
                    <div
                      className="modal-footer"
                    >
                      <button
                        className="btn btn-secondary"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        Cancel
                      </button>
                      <button
                        className="btn btn-primary"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        Submit
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="modal fade"
                id="deleteItemModel1"
                role="dialog"
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <div
                      className="modal-header"
                    >
                      <h5
                        className="modal-title"
                        id="exampleModalLabel"
                      >
                        Delete
                      </h5>
                      <button
                        aria-label="Close"
                        className="close"
                        data-dismiss="modal"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                      </button>
                    </div>
                    <div
                      className="modal-body"
                    >
                      Are you sure you want to delete?
                    </div>
                    <div
                      className="modal-footer"
                    >
                      <button
                        className="btn btn-secondary"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        Cancel
                      </button>
                      <button
                        className="btn btn-primary"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        Delete
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <a
                className="list-group-item list-group-item-action flex-column align-items-start"
                href=""
                onClick={[Function]}
              >
                <div
                  className="container"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-md-2 hidden-sm-down"
                    >
                      <div
                        className="text-center date-display"
                      >
                        <div
                          className="date-display-month text-white bg-cool-blue"
                        >
                           Sep
                        </div>
                        <div
                          className="date-display-day"
                        >
                          <h4>
                             20
                          </h4>
                        </div>
                        <div
                          className="date-display-year"
                        >
                           2017
                        </div>
                      </div>
                    </div>
                    <div
                      className="col-md-10 item-data"
                    >
                      <div
                        className="d-flex w-100 justify-content-between"
                      >
                        <p>
                          <small
                            className="text-muted"
                          >
                            Finished
                          </small>
                        </p>
                        <div>
                          <div
                            className="fa fa-lg fa-pencil-square-o mr-2"
                            data-backdrop="static"
                            data-keyboard="false"
                            data-target="#2"
                            data-toggle="modal"
                            onClick={[Function]}
                          />
                          <div
                            className="fa fa-lg fa-trash-o ml-2"
                            data-backdrop="static"
                            data-keyboard="false"
                            data-target="#deleteItemModel2"
                            data-toggle="modal"
                            onClick={[Function]}
                          />
                        </div>
                      </div>
                      <h5
                        className="mb-1 text-cool-blue"
                      >
                        Test item 2
                      </h5>
                      <h6
                        className="text-muted"
                      >
                        This is a test item 2
                      </h6>
                      <div>
                        <small
                          className="text-muted"
                        >
                          <strong>
                            Updated: 
                          </strong>
                           20 Sep 2017 11:31:50 GMT
                        </small>
                      </div>
                    </div>
                  </div>
                </div>
              </a>
              <div
                className="modal fade"
                id={2}
                role="dialog"
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <div
                      className="modal-header"
                    >
                      <h5
                        className="modal-title"
                        id="exampleModalLabel"
                      >
                        Edit a item
                      </h5>
                      <button
                        aria-label="Close"
                        className="close"
                        data-dismiss="modal"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                      </button>
                    </div>
                    <div
                      className="modal-body"
                    >
                      <label
                        htmlFor="tiltle"
                      >
                        Title
                      </label>
                      <div
                        className="input-group"
                        onChange={[Function]}
                      >
                        <input
                          className="form-control"
                          id="tiltle"
                          name="title"
                          onChange={[Function]}
                          type="text"
                          value="Test item 2"
                        />
                      </div>
                      <br />
                      <label
                        htmlFor="description"
                      >
                        Description
                      </label>
                      <div
                        className="input-group"
                      >
                        <textarea
                          className="form-control"
                          id="description"
                          name="description"
                          onChange={[Function]}
                          type="text"
                          value="This is a test item 2"
                        />
                      </div>
                      <br />
                      <div
                        className="form-group"
                      >
                        <label
                          htmlFor="status"
                        >
                          Status
                        </label>
                        <select
                          className="form-control"
                          id="status"
                          name="status"
                          onChange={[Function]}
                          value="1"
                        >
                          <option
                            value="1"
                          >
                            Finished
                          </option>
                          <option
                            value="0"
                          >
                            Pending
                          </option>
                        </select>
                      </div>
                    </div>
                    <div
                      className="modal-footer"
                    >
                      <button
                        className="btn btn-secondary"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        Cancel
                      </button>
                      <button
                        className="btn btn-primary"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        Submit
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="modal fade"
                id="deleteItemModel2"
                role="dialog"
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <div
                      className="modal-header"
                    >
                      <h5
                        className="modal-title"
                        id="exampleModalLabel"
                      >
                        Delete
                      </h5>
                      <button
                        aria-label="Close"
                        className="close"
                        data-dismiss="modal"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                      </button>
                    </div>
                    <div
                      className="modal-body"
                    >
                      Are you sure you want to delete?
                    </div>
                    <div
                      className="modal-footer"
                    >
                      <button
                        className="btn btn-secondary"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        Cancel
                      </button>
                      <button
                        className="btn btn-primary"
                        data-dismiss="modal"
                        onClick={[Function]}
                        type="button"
                      >
                        Delete
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal fade"
    id="addItemModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="exampleModalLabel"
          >
            Create an item
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          <label
            htmlFor="tiltle"
          >
            Title
          </label>
          <div
            className="input-group"
            onChange={[Function]}
          >
            <input
              className="form-control"
              id="tiltle"
              name="title"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <br />
          <label
            htmlFor="description"
          >
            Description
          </label>
          <div
            className="input-group"
          >
            <textarea
              className="form-control"
              id="description"
              name="description"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <br />
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
          <button
            className="btn btn-primary"
            data-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
